Tensile/KernelWriterAssembly.py

```diff
@@ -2488,6 +2502,20 @@ def functionSignature(self, kernel ):
         kStr += r"    v_add_u32 \dst, vcc, \src0, \src1" + self.endLine
     kStr += ".endm" + self.endLine
 
+    # Use combined shift+or, where available:
+    kStr += "\n"
+    kStr += ".macro _v_lshl_or_b32 dst:req, src0:req, shiftCnt:req, src1:req" + self.endLine
+    if globalParameters["AsmCaps"][self.version]["HasLshlOr"]:
+      kStr += r"    v_lshl_or_b32 \dst, \src0, \shiftCnt, \src1" + self.endLine
+    else:
+      kStr += r"    v_lshlrev_b32 \dst, \shiftCnt, \src0" + self.endLine
+      kStr += r"    v_or_b32 \dst, \dst, \src1" + self.endLine
+      # if self.AsmBugs["ExplicitCO"]:
+      #   kStr += r"    v_add_co_u32 \dst, vcc, \src0, \src1" + self.endLine
+      # else:
+      #   kStr += r"    v_or_b32 \dst, \dst, \src1" + self.endLine
+    kStr += ".endm" + self.endLine
```

Comment:
Not sure about the purpose of ExplicitCO

Reply:
GFX8 has no `v_add_co_u32` instruction. It is implicit in the `v_add_u32`.  GFX9 changed the mnemonic for including carry-out (ExplicitCO).  GFX10 renamed the no-carry case to `v_add_nc_u32` (ExplicitNC).

Reply:
Thanks for the reply, I got it now. So looks like I don't have to care the CO for _v_lshl_or_b32, I'll remove this comment block later.